pkg/provisioner/ironic/ironic.go

```diff
@@ -1225,45 +1224,21 @@ func (p *ironicProvisioner) Deprovision() (result provisioner.Result, err error)
 		"current", ironicNode.ProvisionState,
 		"target", ironicNode.TargetProvisionState,
 		"deploy step", ironicNode.DeployStep,
+		"instance_info", ironicNode.InstanceInfo,
 	)
 
 	switch nodes.ProvisionState(ironicNode.ProvisionState) {
 
 	case nodes.Error, nodes.CleanFail:
-		if !ironicNode.Maintenance {
-			p.log.Info("setting host maintenance flag to force image delete")
-			return p.setMaintenanceFlag(ironicNode, true)
```

Comment:
Clean failure may set `maintenance` mode, you need to clear it here (instead of setting it to `true` which we used to do for some reason).

Reply:
Thanks, I added another commit to do this.